### ISSUE TITLE
Fix unexpected error message in `report()`

### ISSUE THREAD

**Issue by georges-g** [stylelint/stylelint#7172]
### What minimal example or steps are needed to reproduce the bug?

Empty sass file

### What minimal configuration is needed to reproduce the bug?

```json
{
    "customSyntax": "postcss-sass",
    "extends": "stylelint-config-standard"
}
```

### How did you run Stylelint?

```shell
npx stylelint -f verbose "**/*.sass"
```

### Which Stylelint-related dependencies are you using?

```json
  "devDependencies": {
    "postcss-sass": "^0.5.0",
    "stylelint": "^15.10.3",
    "stylelint-config-standard": "^34.0.0"
  }
```

### What did you expect to happen?

No problem to report

### What actually happened?

Stylelint crashed:
```
Error: You must pass either a node or a line number
    at report (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/utils/report.js:42:9)
    at /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/rules/no-empty-source/index.js:32:3
    at /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintPostcssResult.js:121:8
    at Array.map (<anonymous>)
    at lintPostcssResult (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintPostcssResult.js:111:18)
    at lintSource (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintSource.js:110:8)
    at async /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/standalone.js:211:27
    at async Promise.all (index 1)
    at async standalone (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/standalone.js:254:22)
```

### Do you have a proposal to fix the bug?

Stylelint fails to process an empty sass file.

Again, I am not sure if it comes from stylelint or postcss-sass plugin. It works properly with the `postcss-scss` plugin though.

Also, the error message is not very practical, it does not even specify which file made stylelint crash, even with the `-f verbose` option. Any way to improve debugging messages?

**Comment by ybiquitous** [comment]
@georges-g Thanks again for the helpful report. I'm afraid that this also may be a bug of `postcss-sass`. See the following verification:

```console
> require('postcss-scss').parse('').source
{
  input: Input {
    css: '',
    hasBOM: false,
    id: '<input css zG44F1>',
    [Symbol(fromOffsetCache)]: [ 0 ]
  },
  start: { column: 1, line: 1, offset: 0 },
  end: { column: 1, line: 1, offset: 0 }
}
> require('postcss-sass').parse('').source
{
  start: [ 0, 0 ],
  end: [ 0, 0 ],
  input: Input { css: '', hasBOM: false, id: '<input css Ef1dZl>' }
}
```

We can see the difference between the `start` and `end` properties' forms. This form is a [`Position`](https://postcss.org/api/#position) type of PostCSS API. The type is required to have the `column`, `line`, and `offset` properties like `postcss-scss`.

**Comment by georges-g** [comment]
@ybiquitous Thanks again for the useful reply.

I created an [issue there](https://github.com/AleshaOleg/postcss-sass/issues/180).

There is still the issue of stylelint logging and the difficulty to identify which file made it crash. Should we discuss it here or do I create another issue/feature request?

**Comment by georges-g** [comment]
OK, I put this in my todo list, but anyone should feel free to take it. It's a pretty big project and it will take me some time to get to  know the fundamentals of its codebase and contribute appropriately.

**Comment by ybiquitous** [comment]
For example, does the following message satisfy your request?

```
Error: You must pass either a node or a line number (rule: no-empty-source, file: /path/to/foo.sass)
```

If so, we can easily change the error message raised in `report()`.

**Comment by georges-g** [comment]
That would be useful yes.

If you want to go deeper, I think you need several things:
-  If there is not one already, a logging library allowing different log levels
- The option to choose the log level from the CLI
- Some logging instructions at the right places with the right level

Then you will probably not need to customize every `throw` anymore. Easier said than done though, I know ^^

**Comment by ybiquitous** [comment]
Honestly, such a big change is unneeded with this issue.

In addition, since we're mainly developing on the [`v16`](https://github.com/stylelint/stylelint/tree/v16) branch instead of the main branch, we don't hope for a big change.

**Comment by georges-g** [comment]
It might not be so needed for this specific issue, but it could be useful if you look at the bigger picture and consider various debugging issues.

IMO, you could leave this issue open for a while and see what other people think. It could wait for after v16 is released.

If you don't want that, you can go for the quickfix, up to you.
